nemo/collections/asr/parts/preprocessing/perturb.py

```diff
@@ -583,6 +570,204 @@ def perturb_with_foreground_noise(self, data, noise, data_rms=None, max_noise_du
             data._samples[noise_idx : noise_idx + noise_samples.shape[0]] += noise_samples
 
 
+class NoiseNormPerturbation(Perturbation):
+    """
+    Perturbation that adds noise to input audio, with normalisation to specific decibel level.
+    Also tiles shorter noise samples up to their corresponding clean audio length.
+
+    Args:
+        manifest_path (str or list): Manifest file with paths to noise files, can be list if using multiple noise sources
+        min_snr_db (float): Minimum SNR of audio after noise is added
+        max_snr_db (float): Maximum SNR of audio after noise is added
+        snr_samples (list): A discrete list of SNRs DBs to sample from when mixing, will be used instead of [min_snr_db,max_snr_db]
+        norm_to_db (float): Will normalise clean, noise, and mixed samples to this DB
+        audio_tar_filepaths (str or list) : Tar files, if noise audio files are tarred, can be list for multiple sources
+        shuffle_n (int): Shuffle parameter for shuffling buffered files from the tar files
+        orig_sr (int): Original sampling rate of the noise files
+        rng (int): Random seed. Default is None
+        shard_strategy (str): if you're using tarred audio and wish to scatter instead of replicate, set this to 'scatter'
+        epsilon (float): minimum value for RMS DB normalisation to avoid divide by zero
+    """
+
+    def __init__(
+        self,
+        manifest_path=None,
+        min_snr_db=10,
+        max_snr_db=50,
+        snr_samples=None,
+        norm_to_db=None,
+        rng=None,
+        audio_tar_filepaths=None,
+        shuffle_n=128,
+        orig_sr=16000,
+        global_rank=0,
+        world_size=1,
+        shard_strategy='replicate',
+        epsilon=0.01,
+    ):
+        from nemo.collections.asr.data.audio_to_text import RandomizedChainDataset
```

Comment:
Add note that this is to avoid circular import

Reply:
Done